assets/main.js

```diff
@@ -247,3 +247,88 @@ function stickFooterAtBottom() {
 }
 
 stickFooterAtBottom();
+
+/* ------------------------------------------------------ */
+/* ----- Display each video section -------------------- */
+/* ------------------------------------------------------ */
+
+function processVideoSections() {
+  const videoSections = document.querySelectorAll('.video-wrapper');
+
+  if (!videoSections || !videoSections.length) return;
+
+  videoSections.forEach((video) => {
+    const facebookVideo = video.querySelector('.facebook-video');
+    const youtubeVideo = video.querySelector('.youtube-video');
+    const width = video.getAttribute('data-video-width') * (window.innerWidth / 100);
+    const height =  video.getAttribute('data-video-height');
+    const videoUrl=  video.getAttribute('data-video-link');
+
+    function getBestThumbnailUrl(videoId) {
+      return new Promise((resolve) => {
+        const img = new Image();
+        const maxResUrl = `https://i.ytimg.com/vi/${videoId}/maxresdefault.jpg`;
+
+        img.onload = () => {
+          resolve(maxResUrl);
+        };
+
+        img.onerror = () => {
+          resolve(`https://i.ytimg.com/vi/${videoId}/hqdefault.jpg`);
+        };
+
+        img.src = maxResUrl;
+      });
+    }
+
+    function extractYouTubeVideoId(url) {
+      const regex = /(?:v=|youtu\.be\/)([^&]+)/;
+      const match = url.match(regex);
+      return match ? match[1] : null;
+    }
+
```

Comment:
The video section code needs to be on the video.js file.There is no need to include the js file in the main sinnce this JS code is only required on the home page.

Reply:
We need to put it inside the main js because if we put it inside  video.js file this file is included in the video section so each time we add the video section we will have duplicated js code this isn't optimal for better performance and clean code